src/test/scala/com/strad/service/ServiceRoutesSpec.scala

```diff
@@ -0,0 +1,62 @@
+package com.strad.service
+
+import cats.*
+import cats.effect.*
+import cats.implicits.*
+import com.strad.ServiceRoutes
+import com.strad.repositories.UserRepo
+import com.strad.repositories.domain.User
+import com.strad.service.ModelsJson.given
+import com.strad.service.domain.UserResponse
+import io.circe.*
+import io.circe.generic.auto.*
+import io.circe.syntax.*
+import munit.CatsEffectSuite
+import org.http4s.*
+import org.http4s.circe.*
+import org.http4s.circe.CirceEntityDecoder.circeEntityDecoder
+import org.http4s.client.Client
+import org.http4s.dsl.io.*
+import org.http4s.implicits.uri
+
+class ServiceRoutesSpec extends CatsEffectSuite:
+  private val success: UserRepo[IO] = new UserRepo[IO]:
+    override def getUserById(id: Long): IO[Option[User]] = IO.pure(Some(User(123, None)))
+    override def insertUser(user: User): IO[User] = ???
+    override def deleteUserById(userId: Long): IO[Unit] = ???
+
+  private val foundNone: UserRepo[IO] = new UserRepo[IO]:
+    override def getUserById(id: Long): IO[Option[User]] = IO.pure(None)
+    override def insertUser(user: User): IO[User] = ???
+    override def deleteUserById(userId: Long): IO[Unit] = ???
+
+  private val exception: UserRepo[IO] = new UserRepo[IO]:
+    override def getUserById(id: Long): IO[Option[User]] = IO.raiseError(new RuntimeException("Should not get called!"))
+    override def insertUser(user: User): IO[User] = ???
+    override def deleteUserById(userId: Long): IO[Unit] = ???
+
+  private def userServiceRoutes(repo: UserRepo[IO]) =
+    ServiceRoutes.userRoutes[IO](new UserServiceImpl[IO](repo)).orNotFound
+
+  test("GET /api/v1/users/123 should return 200 OK with the correct response"):
```

Comment:
this is how to get rid of the main method you had by using tests.  You also created a check method which we could add back but for now felt this was more straightforward to see what is going on.